platform/packages/finance/src/coin/dto/amount_serde.rs

```diff
@@ -0,0 +1,105 @@
+use std::fmt::Formatter;
+
+/// This module implements (de-)serialation of Amount as String
+/// to keep compatibility with pre-CW 2.x
+use serde::{
+    de::{Unexpected, Visitor},
+    Deserializer, Serializer,
+};
+
+use crate::coin::Amount;
+
+pub(super) fn serialize<S>(amount: &Amount, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    serializer.serialize_str(&amount.to_string())
+}
+
+pub(super) fn deserialize<'de, D>(deserializer: D) -> Result<Amount, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct StrVisitor();
```

Comment:
[Style] No need for the parenthesis.
```suggestion
    struct StrVisitor;
```